qrcode_app/lib/data/models/seed_model.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:qrcode_app/domain/entities/seed.dart';
+
+class SeedModel {
+  late String seed;
+  late DateTime expirationDate;
+
+  Seed fromJson(dynamic json) {
+    seed = json['seed'] ?? 'default';
+    // expirationDate = DateTime.now().add(Duration(seconds: 10));
+    expirationDate = DateTime.tryParse(json['expiration']) ??
+        DateTime.now().add(Duration(hours: 1));
+    return Seed(expirationDate: expirationDate, seed: seed);
+  }
+}
```

Comment:
**chore**: Since the SeedModel (Dto) and Seed live in different layers there should be a mapper between layers.

In data layer...

```
import 'package:qrcode_app/data/models/seed_model.dart';
import 'package:qrcode_app/domain/entities/seed.dart';

class SeedMapper {
  Seed toSeed(SeedDto dto) => Seed(seed:dto.seed, expirationDate: dto.expirationDate);
}

```
If you were sending requests this would contain the mapper the other way (toDto) as well.

Reply:
Thank you so much for the tips. I appreciate it. I will change it!